src/mac.m

```diff
@@ -1174,23 +1174,21 @@ - (void)windowDidResignKey:(NSNotification*)notification
 puglSendEvent(PuglView* view, const PuglEvent* event)
 {
   if (event->type == PUGL_CLIENT) {
-    PuglWrapperView* wrapper = view->impl->wrapperView;
-    const NSWindow*  window  = [wrapper window];
-    const NSRect     rect    = [wrapper frame];
-    const NSPoint    center  = {NSMidX(rect), NSMidY(rect)};
-
-    NSEvent* nsevent =
-      [NSEvent otherEventWithType:NSApplicationDefined
-                         location:center
-                    modifierFlags:0
-                        timestamp:[[NSProcessInfo processInfo] systemUptime]
-                     windowNumber:window.windowNumber
-                          context:nil
-                          subtype:PUGL_CLIENT
-                            data1:(NSInteger)event->client.data1
-                            data2:(NSInteger)event->client.data2];
-
-    [view->world->impl->app postEvent:nsevent atStart:false];
+    PuglEvent copiedEvent = *event;
+
+    CFRunLoopObserverRef observer = CFRunLoopObserverCreateWithHandler(
+      NULL,
+      kCFRunLoopAfterWaiting,
+      false,
+      0,
+      ^(CFRunLoopObserverRef, CFRunLoopActivity) {
+        puglDispatchEvent(view, &copiedEvent);
```

Comment:
Isn't this referring to garbage on the stack (copiedEvent)?

Reply:
It doesn't look that way in a debugger.

According to [this page](https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/WorkingwithBlocks/WorkingwithBlocks.html#//apple_ref/doc/uid/TP40011210-CH8-SW3), referenced objects are copied unless you specify otherwise.